newdevtech/newdev_api
# #166487874 Implement Create Read operations

#### What does this PR do?
- adds controller for rating
- adds middleware for rating
- adds route for rating
- refactors typeValdator for Array and number types
- refactors user-ratings-resource model associations
- renames review test file

#### Description of Task to be completed?
- We need to implement the user-resource-rating functionality so that users can successfully rate articles

#### How should this be manually tested?
- Send a POST request to `/api/v1/ratings` with the required params: `resourceId` and `rating`
  - You should receive a success response if params are valid
- Send a GET request to `/api/v1/ratings/:resourceId`
  - You should receive a success response with `averageRating` if the resource exists

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166487874](https://www.pivotaltracker.com/story/show/166487874)
